crates/bevy_core/src/time/time.rs

```diff
@@ -1,26 +1,49 @@
 use bevy_ecs::system::ResMut;
 use bevy_utils::{Duration, Instant};
 
-/// Tracks elapsed time since the last update and since the App has started
+use crate::FixedTimestepState;
+
+/// Tracks time elapsed since the previous update and since the app was started.
 #[derive(Debug, Clone)]
 pub struct Time {
     delta: Duration,
-    last_update: Option<Instant>,
-    delta_seconds_f64: f64,
     delta_seconds: f32,
-    seconds_since_startup: f64,
+    delta_seconds_f64: f64,
+    raw_delta: Duration,
+    raw_delta_seconds: f32,
+    raw_delta_seconds_f64: f64,
+    relative_speed: f32,
+    elapsed_since_startup: Duration,
+    first_update: Option<Instant>,
+    last_update: Option<Instant>,
     startup: Instant,
+    fixed_delta: Duration,
+    fixed_delta_seconds: f32,
+    fixed_delta_seconds_f64: f64,
+    fixed_elapsed_since_startup: Duration,
 }
 
 impl Default for Time {
-    fn default() -> Time {
-        Time {
-            delta: Duration::from_secs(0),
+    fn default() -> Self {
+        // 60Hz is a popular number, but it can't be expressed as an exact float.
+        // Powers of two (i.e. 32Hz, 64Hz, 128Hz) are better for numerical integration.
+        let fixed_delta = Duration::from_secs_f32(1.0 / 64.0);
+        Self {
+            delta: Duration::ZERO,
+            delta_seconds: 0.0,
+            delta_seconds_f64: 0.0,
+            raw_delta: Duration::ZERO,
+            raw_delta_seconds: 0.0,
+            raw_delta_seconds_f64: 0.0,
+            relative_speed: 1.0,
+            elapsed_since_startup: Duration::ZERO,
+            first_update: None,
             last_update: None,
             startup: Instant::now(),
-            delta_seconds_f64: 0.0,
-            seconds_since_startup: 0.0,
-            delta_seconds: 0.0,
+            fixed_delta,
+            fixed_delta_seconds: fixed_delta.as_secs_f32(),
+            fixed_delta_seconds_f64: fixed_delta.as_secs_f64(),
```

Comment:
Which fields are redundantly stored and which are not feels a little arbitrary here.

Reply:
You're talking about how the `Duration`, `f32`, and `f64` values are stored for delta time but only `Duration` for elapsed time, right?

The redundancy avoids paying the small cost of [converting a `Duration` to a float][1] each time you want to read the value. I think I just assumed that delta time would be read way more often than elapsed time, but we can definitely add the float fields for the elapsed times.

[1]: https://doc.rust-lang.org/stable/src/core/time.rs.html#684

Reply:
Fair point, you are right that persisting the values makes sense to some degree. Although I do wonder if both f32 and f64 is necessary.

Reply:
Based on <https://github.com/bevyengine/bevy/pull/3264#issuecomment-987476724>, I opted to persist everything.